### ISSUE TITLE
ECHOConfigContents.extensions is not needed

### ISSUE THREAD

**Issue by sftcd** [tlswg/draft-ietf-tls-esni#217]
draft-06 includes an extensions field in ECHOConfigContents but no extension types are defined. If or when something new is needed it could just as easily be introduced via a new ECHOConfig.version or a new HTTPSSVC tag. The mandatory-to-understand design wit a "critical" bit was also used in X.509 and has not clearly worked as planned there. I don't see why that would differ here. Expecting implementers to include all the code required for generic extension handling and introducing new failure modes via critical bits, all for no concrete reason... seems wrong. I suggest we just delete the extensions field entirely. If needed, one can be added back via a new version later. 

Bearing in mind the target for this draft is an experimental RFC, and so there's a reasonable likelihood of changes when/if we try promote ECHO to a standards-track RFC, we also don't need to be quite as future proof as normal, so deleting this field for now is not that risky.

**Comment by ekr** [comment]
I don't agree with this proposed change.

1.  Extensions for TLS have proved very useful.
2. The analogy to X.509 seems false. Criticality has a different context here and IMO it works fine for (say) STUN.

Introducing a new version has all kinds of other annoyances.

As for the experimental bit, that seems like an oversight. I expected this to be PS and I'm not sure why the draft has a different tag. I expect I just screwed up when I wrote the first draft. I note the adoption call doesn't say one way or the other. As a practical matter, we expect this to have widespread deployment.
